### DISCUSSION TITLE
Error in a model in MoBi

### DISCUSSION THREAD

**Discussion by MaLuisaEti** [Open-Systems-Pharmacology/Forum#1235] [Models]
Hello,
We are working in a PBPK model for a drug which causes renal vasodilatation and it has a renal elimination. We are trying to optimize our model in MoBi considering the effect of plasma concentration of the drug on its renal excretion by the change in the renal blood flow rate.
We found an equation that estimates the Specific blood flow rate as a function to our drug's concentration. So in order to implement it in our model, we change the Parameter **Specific blood flow rate** from _Formula type: Constant_ to _Formula type: Formula (an explicit formula)_ and we wrote the equation linking the specific blood flow rate to the total drug concentration. 

We previously added the total venous concentration of the drug as a new parameter to use it as part of our equation that estimates the Specific blood flow rate, and we did it in the following way:
1) ‘add parameter’ of type: Global, Dimension: Concentration, Group: Mobi, Formula type: Formula (an explicit formula)
2) Then right click in the white space prepared for adding equation parameters and ‘add reference’ and there included the total venous blood concentration using the path: Organism|VenousBlood|Plasma|MOLECULE|Concentration

(we follow the step by step of discussion #907)

Then, when we update our simulation with the new version of our spatial structure but the following error appears:
![Error_Forum](https://user-images.githubusercontent.com/70070603/175121839-4cdcbaed-815c-453d-9d6c-a343af258b10.png)

We don't know what it means or how to fix this error. If anyone can help us, we will appreciate it very much!

Thank you in advance
María Luisa

**Comment by msevestre** [comment]
@MaLuisaEti I cannot read the message. Can you upload a higher resolution? 
Also maybe upload the mobi file if possible (or a modified version/simplified) where we can reproduce the problem.

**Reply by MaLuisaEti** [reply]
Hi Michael, 

Yes, here the message with higher resolution:
![Error_Forum2](https://user-images.githubusercontent.com/70070603/175142629-f091667a-e78f-4ca4-bec3-1d64848c93b7.png)

And the mobi file simplified:

[Qspec_Cdepend - simpl.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/8961578/Qspec_Cdepend.-.simpl.zip)

Thanks a lot!
Ma Luisa

**Comment by StephanSchaller** [comment]
Hi @MaLuisaEti , you would have to define this parameter `Cfuro` in a `Molecule properties` Container. E.g.: ..|Kidney|Plasma|Molecule Properties:
![image](https://user-images.githubusercontent.com/25477643/175315589-416c90b3-be05-4807-b074-1718d9f08bbd.png)
Otherwise the keyword `MOLECULE` cannot get assigned.
If it is a global parameter, you can also define it at the root level:
![image](https://user-images.githubusercontent.com/25477643/175315282-10d8e369-215e-4c4e-828a-753efdf1bad1.png)

(Don't forget to adapt the path to the parameter wherever it is used)

Let me know if this works!

Stephan

**Reply by MaLuisaEti** [reply]
Hi Stephan,
Thanks for your response. I defined the parameter Cfuro in the Molecule Properties in the container Kidney|Plasma but unfortunately, I haven´t had success and the same error appears. Cfuro is defined at the root level too.
Maybe I´m not adapting the path correctly, but I think that at the root level has to be:  Organism|VenousBlood|Plasma|MOLECULE|Concentration, is that correct?

Thank you again
María Luisa

**Reply by StephanSchaller** [reply]
Okay, I saw it on the `Molecule Propertis` on the root level.
So, in your attached *.zip, when I delete the (wrong) `Cfuro` in Kidney, I can update the simulation without any error message. So the `Cfuro` in `Molecule Propertis` on the root level works fine.

**Reply by MaLuisaEti** [reply]
Yes Stephan, you are right. However, the Parameter **Specific blood flow rate** that is in the container Kidney and it is a function on the total drug concentration (Cfuro) (we change the Parameter ¨Specific blood flow rate¨ from ¨Formula type: Constant¨ to ¨Formula type: Formula (an explicit formula)¨), seems not to be affected by the parameter Cfuro when is only at the root level. At least when we run the simulation, the Parameter **Specific blood flow rate** remains as a constant. Any clue about this? If Cfuro is on the root level, it should be enough, right?

Thank you for your time!
María Luisa

**Reply by StephanSchaller** [reply]
@MaLuisaEti, can I ask why you want to use the keyword `MOLECULE`? You know your drug name (furosemide?) so you could just use it explicitly (as blood flow is only affected by this one drug), then you do not have this issue. Just use C_plasma directly as by definition it is >= 0. C_plasma would then have the path `.|Plasma|Furosemide|Concentration`

**Reply by MaLuisaEti** [reply]
Hello Stephan, I thought I had to use the keyword MOLECULE. But now I understand what you are pointing out. 

Update: I could resolve my problem 🙂. It was necessary to change the parameter Specific blood flow rate in the buiding block Parameter Start Values as well as in Spatial Structures otherwise the parameter did not respond to the indicated equation. 

Thank you
María Luisa

**Reply by StephanSchaller** [reply]
Yes, the PSV update is a noteworthy step that is often overlooked!! Thanks for posting your solution!
